yt_dlp/extractor/spreaker.py

```diff
@@ -94,28 +95,33 @@ class SpreakerIE(InfoExtractor):
     }]
 
     def _real_extract(self, url):
-        episode_id = self._match_id(url)
+        episode_id, key = self._match_valid_url(url).group(
+            'id', 'key')
+        query = {}
+        if key is not None:
+            query = {'key': key}
```

Comment:
```suggestion
        episode_id = self._match_id(url)
        query = {}
        if key := traverse_obj(url, ({parse_qs}, 'key', 0)):
            query = {'key': key}
```

import `traverse_obj` from `..utils.traversal` and `parse_qs` from `..utils`